estimagic/optimization/check_constraints.py

```diff
@@ -16,6 +16,9 @@ def check_constraints_are_satisfied(pc, params):
     This should be called before the more specialized constraints are rewritten to
     linear constraints in order to get better error messages!
 
+    We let the checks pass if all parameter values are np.nan. This is useful for the
+    construction of star parameters.
```

Comment:
What are star parameters?

Reply:
They are star**t** parameters without a t in the end ...